File: Fase1/2015/nivel1/linhas_cruzadas.py
N = int(input()) #pegamaos o numero de pregos
pregos = input().split() #pegamos a ordem na linha horizontal dos pregos

#Aqui convertemos os valores da lista para inteiros
pregos_num = []
for p in pregos:
    num_p = int(p)
    pregos_num.append(num_p)

cruzamentos = 0 #Uma variavel para contar os cruzamentos

'''
Basicamente, vamos percorrer a lista, e para cada valor, verificamos se ele cruza com 
os outros numeros, por exemplo, se pegarmos o 5, verificamos se os numeros afrente dele na lista são 
menores que ele, caso sejam, há um cruzamento. Assim, cada numero é analisado.

'''
for i in range(N):
    for j in range(i+1, N):
        if(pregos_num[j] < pregos_num[i]):
            cruzamentos += 1

'''
N = 6
pregos_num = [1, 5, 3, 4, 6, 2]
cruzamentos = 3

i = 2
j = 3

4 < 3
'''

print(cruzamentos)

File: Fase1/2015/nivel1/linhas_cruzadas.cpp
#include <iostream>
#include <vector>
#include <string>
#include <sstream>

using namespace std;

int main() {
    int N;
    cout << "Digite o número de pregos: ";
    cin >> N;

    cin.ignore(); // Limpar o buffer do newline após a entrada do número

    vector<int> pregos_num;
    cout << "Digite a ordem dos pregos na linha horizontal separados por espaço: ";
    string input;
    getline(cin, input);
    istringstream iss(input);
    int num_p;
    while (iss >> num_p) {
        pregos_num.push_back(num_p);
    }

    int cruzamentos = 0;

    for (int i = 0; i < N; i++) {
        for (int j = i + 1; j < N; j++) {
            if (pregos_num[j] < pregos_num[i]) {
                cruzamentos++;
            }
        }
    }

    cout << "O número de cruzamentos é: " << cruzamentos << endl;

    return 0;
}
